File: backend/src/008-importar_modulos/ruby/main.rb
#!/usr/bin/ruby

# ---------  IMPORTAR EN EL MISMO DIRECTORIO  -----------------

# require_relative permite importar módulos relativos al archivo
# donde se ha declarado esta sentencia
require_relative "modulo"

# ---------  IMPORTAR DE OTROS DIRECTORIOS  -------------------
require "./directorio/modulo" # Así también --> "./directorio/modulo.rb"
# Sin require_relative tenemos que escribir "./" antes del nombre
# si el módulo se encuentra en el mismo directorio


=begin
Otra forma de importar es usar  -->  load "./modulo"
En este caso, el modulo volverá a cargarse,
si han ocurrido cambios en su interior podremos
acceder a ellos
=end

if __FILE__ == $0  # Equivalente en python -> if __name__ == "__main__":
    funcion
    funcion_otro_directorio
end

=begin
Fuentes:
https://stackoverflow.com/questions/3672586/what-is-the-difference-between-require-relative-and-require-in-ruby
=end

File: backend/src/008-importar_modulos/c++/main.cpp
#include <iostream>
using namespace std;

int main(){
    // INCLUIR DEL MISMO DIRECTORIO
    #include "modulo.cpp"    // cout << "Desde otro fichero en "
    #include "modulo2.cpp"   // "el mismo directorio" << endl;
    
    // INCLUIR DESDE OTRO DIRECTORIO
    #include "directorio/modulo.cpp"
    
    return 0;
}

/* Fuente:
https://trucosinformaticos.wordpress.com/2012/10/14/como-usar-include-en-c-y-c/
*/
